### PR TITLE
(CRO-5 CRO-68) Crypto.com Chain Client: Wallet implementation

### PR DESCRIPTION
**Pull request by devashishdxt**

### CODE REVIEW

File: client-core/src/key.rs [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+    /// Generates a new private key
+    pub fn new() -> Result<PrivateKey> {
+        let mut rng = OsRng::new().context(ErrorKind::KeyGenerationError)?;
+        let secret_key = SecretKey::new(&mut rng);
+
+        Ok(PrivateKey(secret_key))
+    }
+
+    /// Serializes current private key
+    pub fn serialize(&self) -> Result<Vec<u8>> {
```
Comment (tomtau) [inline client-core/src/key.rs]: are these for import/export?

File: client-core/src/key.rs [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+
+    /// Deserializes private key from bytes
+    pub fn deserialize_from(bytes: &[u8]) -> Result<PrivateKey> {
+        let secret_key: SecretKey = deserialize(bytes).context(ErrorKind::DeserializationError)?;
+
+        Ok(PrivateKey(secret_key))
+    }
+}
+
+impl Zeroize for PrivateKey {
```
Comment (tomtau) [inline client-core/src/key.rs]: perhaps `impl Drop for PrivateKey` that would call zeroize?

File: client-core/src/wallet.rs [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+
+                private_key.zeroize();
+
+                Ok(public_key)
+            }
+        }
+    }
+
+    /// Generates a new address
+    fn generate_address(&self, name: &str, passphrase: &str) -> Result<String> {
```
Comment (tomtau) [inline client-core/src/wallet.rs]: What is this one used for?

File: client-core/src/lib.rs [hunk 4]
```diff
@@ -0,0 +1,35 @@
+#![deny(missing_docs, unsafe_code, unstable_features)]
+//! # Crypto.com Chain Client
+//!
+//! This crate exposes following functionalities for interacting with Crypto.com Chain:
+//! - Wallet creation
+//! - Address generation
+//! - Balance tracking
```
Comment (tomtau) [inline client-core/src/lib.rs]: I suppose balance tracking is not implemented yet?

Comment (devashishdxt) [reply client-core/src/lib.rs]: No. Right now, only wallet creation and address generation are implemented. I'm still working on those features and also, a PR with all the features will become too big to review.

File: client-core/src/wallet.rs (hunk 3 above)
Comment (devashishdxt) [reply client-core/src/wallet.rs]: This will be used by CLI / RPC to generate new addresses for a wallet.

**Pull request merged**